proc/groupby.go

```diff
@@ -125,16 +134,54 @@ type GroupByAggregator struct {
 	// lookup table so that values with the same encoding but of
 	// different types do not collide.  No types from this context
 	// are ever referenced.
-	kctx          *resolver.Context
-	keys          []GroupByKey
-	reducerDefs   []compile.CompiledReducer
-	builder       *ColumnBuilder
-	table         map[string]*GroupByRow
-	limit         int
-	valueCompare  expr.ValueCompareFn // to compare primary group keys for early key output
-	recordCompare expr.CompareFn
-	maxKey        *zng.Value
-	inputSortDir  int
+	kctx         *resolver.Context
+	keys         []GroupByKey
+	decomposable bool
+	reducerDefs  []compile.CompiledReducer
+	builder      *ColumnBuilder
+	table        map[string]*GroupByRow
+	limit        int
+	valueCompare expr.ValueCompareFn // to compare primary group keys for early key output
+	keyCompare   expr.CompareFn      // compare the first key (used when input sorted)
+	keysCompare  expr.CompareFn      // compare all keys
+	maxKey       *zng.Value
+	inputSortDir int
+	spillManager *spillManager
+	combiner     *Combiner
+}
+
+type spillManager struct {
+	tempDir string
+	n       int
+}
+
+func newSpillManager() (*spillManager, error) {
```

Comment:
I don't see a call to remove the temporary directory created here at the end of the groupby operation.

Reply:
Oops, I had this (and other cleanup) in a stash that i forgot to apply before putting up the PR.  c1a14ba